packages/stark-ui/src/modules/message-pane/services/message-pane.service.spec.ts

```diff
@@ -0,0 +1,210 @@
+	let mockStore: SpyObj<Store<StarkUIApplicationState>>;
+	const mockLogger: MockStarkLoggingService = new MockStarkLoggingService();
+	let appMessagesState$: BehaviorSubject<StarkMessageCollection>;
+	let mockMessageCollection: StarkMessageCollection;
+	let messagePaneService: MessagePaneServiceHelper;
+
+	const mockInfoMessage: StarkMessage = {
+		id: "3",
+		key: "MESSAGES.INFOS.DUMMY",
+		interpolateValues: { var1: "some info value", var2: "whatever" },
+		code: "1234",
+		type: StarkMessageType.INFO
+	};
+	const mockWarningMessage: StarkMessage = {
+		id: "2",
+		key: "MESSAGES.WARNINGS.DUMMY",
+		interpolateValues: { var1: "some warning value", var2: "whatever" },
+		code: "4321",
+		type: StarkMessageType.WARNING
+	};
+	const mockErrorMessage: StarkMessage = {
+		id: "1",
+		key: "MESSAGES.ERRORS.DUMMY",
+		interpolateValues: { var1: "some error value", var2: "whatever" },
+		code: "0159",
+		type: StarkMessageType.ERROR
+	};
+
+	beforeEach(() => {
+		mockStore = jasmine.createSpyObj<Store<StarkUIApplicationState>>("store", ["dispatch", "pipe"]);
+
+		mockMessageCollection = {
+			errorMessages: [],
+			warningMessages: [],
+			infoMessages: []
+		};
+
+		appMessagesState$ = new BehaviorSubject(mockMessageCollection);
+		(<Spy>mockStore.pipe).and.returnValue(appMessagesState$);
+
+		messagePaneService = new MessagePaneServiceHelper(mockLogger, mockStore);
+	});
+
+	afterEach(() => {
+		appMessagesState$.complete();
+	});
+
+	describe("on initialization", () => {
+		it("should subscribe to the store to fetch the latest value of the starkMessages (StarkMessageCollection)", () => {
+			expect(mockStore.pipe).toHaveBeenCalledTimes(1);
+			expect(messagePaneService.messages$).toBeDefined();
+
+			const mockMessagesObserver: Observer<any> = createSpyObj<Observer<any>>("observerSpy", ["next", "error", "complete"]);
+
+			messagePaneService.messages$.subscribe(mockMessagesObserver);
+
+			expect(mockMessagesObserver.next).toHaveBeenCalledTimes(1);
+			expect(mockMessagesObserver.next).toHaveBeenCalledWith(mockMessageCollection);
+			expect(mockMessagesObserver.error).not.toHaveBeenCalled();
+			expect(mockMessagesObserver.complete).not.toHaveBeenCalled();
+		});
+	});
+
+	describe("add", () => {
+		it("should dispatch the ADD_MESSAGES action passing the given messages in the payload", () => {
+			const mockMessages: StarkMessage[] = [mockInfoMessage, mockWarningMessage, mockErrorMessage];
+
+			messagePaneService.add(mockMessages);
+
+			mockMessages[0] = { ...mockInfoMessage, priority: 999 };
+			mockMessages[1] = { ...mockWarningMessage, priority: 999 };
+			mockMessages[2] = { ...mockErrorMessage, priority: 999 };
+
+			expect(mockStore.dispatch).toHaveBeenCalledTimes(1);
+			expect(mockStore.dispatch).toHaveBeenCalledWith(new StarkAddMessages(mockMessages));
+		});
+	});
+
+	describe("addOne", () => {
+		it("should dispatch the GET_ALL_MESSAGES action and return the observable used to fetch the latest value of the starkMessages", () => {
```

Comment:
"should dispatch the GET_ALL_MESSAGES action..." -> "should dispatch the ADD_MESSAGES action..."